packages/flutter_tools/test/general.shard/pub_dependencies_project_validator_test.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/project_validator.dart';
+import 'package:flutter_tools/src/project_validator_result.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+
+void main() {
+  late FileSystem fileSystem;
+
+  group('PubDependenciesProjectValidator', () {
+
+    setUp(() {
+      fileSystem = globals.localFileSystem;
+    });
+
+    testWithoutContext('success ', () async {
+      final ProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+        const FakeCommand(
+          command: <String>['dart', 'pub', 'deps', '--json'],
+          stdout: '{"packages": [{"dependencies": ["abc"], "source": "hosted"}]}',
+        ),
+      ]);
+      final PubDependenciesProjectValidator validator = PubDependenciesProjectValidator(processManager);
+
+      final List<ProjectValidatorResult> result = await validator.start(
+          FlutterProject.fromDirectoryTest(fileSystem.currentDirectory)
+      );
+      const String expected = 'All dependencies are hosted';
```

Comment:
will the user have enough context to understand what this means?

Reply:
The user will read `[✓] Dart dependencies: All dependencies are hosted`

Reply:
will they know what that means?

Reply:
I don't know. Should I add something like `[✓] Dart dependencies: All dart dependencies comes from known places` or something along those lines?